### PR TITLE
fix and improve

### PR DESCRIPTION
**Pull request by superboyiii**
Close https://github.com/neo-project/neo-node/issues/784

### CODE REVIEW

File: neo-cli/CLI/MainService.Vote.cs [hunk 1]
```diff
@@ -7,22 +7,21 @@
...
         /// Process "register candidate" command
         /// </summary>
         /// <param name="account">register account scriptHash</param>
         /// <param name="maxGas">Max fee for running the script</param>
         [ConsoleCommand("register candidate", Category = "Vote Commands")]
-        private void OnRegisterCandidateCommand(UInt160 account, decimal maxGas = RegisterGas)
+        private void OnRegisterCandidateCommand(UInt160 account)
         {
-            var gas = new BigDecimal(maxGas, NativeContract.GAS.Decimals);
+            var gas = NativeContract.NEO.GetRegisterPrice(NeoSystem.StoreView) + (BigInteger)Math.Pow(10, NativeContract.GAS.Decimals) * 10;
```
Comment (shargon) [inline neo-cli/CLI/MainService.Vote.cs:24]: TestGas?

**Comment by nicolegys** [comment]
If the account don't have any neo
image

**Review by nicolegys — APPROVED** [review]

**Comment by superboyiii** [comment]
@shargon Could you review again?

File: neo-cli/CLI/MainService.Vote.cs (hunk 1 above)
Comment (shargon) [reply neo-cli/CLI/MainService.Vote.cs]: I mean use the test gas variable, no change the name

**Comment by superboyiii** [comment]
@erikzhang Could you help review this?

File: neo-cli/CLI/MainService.Vote.cs (hunk 1 above)
Comment (superboyiii) [reply neo-cli/CLI/MainService.Vote.cs]: > I mean use the test gas variable, no change the name

TestGas can be not enough for dynamic RegisterPrice. We should get it from NEO.GetRegisterPrice().

File: neo-cli/CLI/MainService.Vote.cs [hunk 2]
```diff
@@ -219,21 +218,31 @@ private void OnGetNextBlockValidatorsCommand()
...
             var arg = new JObject();
             arg["type"] = "Hash160";
             arg["value"] = address.ToString();
 
             if (!OnInvokeWithResult(NativeContract.NEO.Hash, "getAccountState", out StackItem result, null, new JArray(arg))) return;
-
+            Console.WriteLine();
+            if (result.IsNull)
+            {
+                Console.WriteLine(notice);
```
Comment (shargon) [inline neo-cli/CLI/MainService.Vote.cs:230]: notice variable is not needed

Comment (superboyiii) [reply neo-cli/CLI/MainService.Vote.cs:230]: > notice variable is not needed

There're two places need this msg. Do you mean repeat it twice straightly in Console.WriteLine()?

**Comment by erikzhang** [comment]
Test?

**Review by erikzhang — APPROVED** [review]

**Review by nicolegys — APPROVED** [review]

**Pull request merged**